geth-patches/0004-protect.patch

```diff
@@ -0,0 +1,23 @@
+diff --git a/internal/ethapi/backend.go b/internal/ethapi/backend.go
+index ccc05ecbd..9ab88c906 100644
+--- a/internal/ethapi/backend.go
++++ b/internal/ethapi/backend.go
+@@ -120,6 +120,7 @@ func GetAPIs(apiBackend Backend, chain *core.BlockChain) []rpc.API {
+ 			Service:   NewPersonalAccountAPI(apiBackend, nonceLock),
+ 		}, {
+ 			Namespace: "eth",
++			Version:   "protect",
+ 			Service:   NewPrivateTxBundleAPI(apiBackend),
+ 		}, {
+ 			Namespace: "eth",
+diff --git a/node/node.go b/node/node.go
+index b60e32f22..3969276ac 100644
+--- a/node/node.go
++++ b/node/node.go
+@@ -574,5 +574,5 @@ func (n *Node) RegisterAPIs(apis []rpc.API) {
+ 	for _, api := range n.rpcAPIs {
+-		if !api.Authenticated {
++		if !api.Authenticated && api.Version == "protect" {
```

Comment:
is that only for the SSL API, or is it restricting all APIs?

Reply:
This applies to all non authenticated endpoints. This patch can be combined with the TLS patch, [which wraps all non authenticated endpoints in a TLS session](https://github.com/flashbots/geth-sgx-gramine/blob/main/geth-patches/0003-go-ethereum-tls.patch#L16-L23).

Reply:
So to be clear, the `--ws` and `--http` endpoints will be restricted to `eth_sendBundle` only.

Reply:
Sounds good 👍 

Should we mention this in the README, that the patched builder only supports the sendBundle API call?

Reply:
@metachris this feature is hidden behind a build flag, which is documented in the readme here: https://github.com/flashbots/geth-sgx-gramine/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R143